packages/thegraph-data-access/src/subgraph-client.ts

```diff
@@ -1,51 +1,110 @@
-import { DataAccessTypes } from '@requestnetwork/types';
+import { DataAccessTypes, StorageTypes } from '@requestnetwork/types';
 import { GraphQLClient } from 'graphql-request';
 import {
   GetBlock,
   GetChannelsByTopicsQuery,
   GetTransactionsByChannelIdQuery,
   GetTransactionsByHashQuery,
   Meta,
+  Transaction,
   TransactionsBody,
 } from './queries';
 import { RequestInit } from 'graphql-request/dist/types.dom';
+import { Variables } from 'graphql-request/dist/types';
 
 // Max Int value (as supported by grapqhl types)
 const MAX_INT_VALUE = 0x7fffffff;
 
-export class SubgraphClient {
+export class SubgraphClient implements StorageTypes.IIndexer {
   private graphql: GraphQLClient;
   public readonly endpoint: string;
   constructor(endpoint: string, options?: RequestInit) {
     this.endpoint = endpoint;
     this.graphql = new GraphQLClient(endpoint, options);
   }
 
+  public async initialize(): Promise<void> {
+    await this.getBlockNumber();
+  }
+
   public async getBlockNumber(): Promise<number> {
     const { _meta } = await this.graphql.request<Meta>(GetBlock);
     return _meta.block.number;
   }
 
-  public getTransactionsByHash(hash: string): Promise<TransactionsBody> {
-    return this.graphql.request<TransactionsBody>(GetTransactionsByHashQuery, {
-      hash,
-    });
+  public getTransactionsByStorageLocation(
+    hash: string,
+  ): Promise<StorageTypes.IGetTransactionsResponse> {
+    return this.fetchAndFormat(GetTransactionsByHashQuery, { hash });
   }
 
   public getTransactionsByChannelId(
     channelId: string,
     updatedBetween?: DataAccessTypes.ITimestampBoundaries,
-  ): Promise<TransactionsBody> {
-    return this.graphql.request<TransactionsBody>(GetTransactionsByChannelIdQuery, {
+  ): Promise<StorageTypes.IGetTransactionsResponse> {
+    return this.fetchAndFormat(GetTransactionsByChannelIdQuery, {
       channelId,
       ...this.getTimeVariables(updatedBetween),
     });
   }
 
-  public getChannelsByTopics(topics: string[]): Promise<TransactionsBody> {
-    return this.graphql.request<TransactionsBody>(GetChannelsByTopicsQuery, {
-      topics,
-    });
+  // FIXME: this should be possible to do in a single query to the subgraph,
+  // but currently one transaction doesn't contain topics from previous ones on the same channel.
+  // This could be fixed on the Subgraph indexer code for optimization.
```

Comment:
👍 Very clear, thanks for documenting the fixme

Reply:
Issue https://github.com/RequestNetwork/storage-subgraph/issues/9